cmd/influx/task.go

```diff
@@ -3,342 +3,455 @@ package main
+	if b.org.name != "" {
+		filter.Organization = b.org.name
 	}
-	if taskFindFlags.org.id != "" {
-		id, err := influxdb.IDFromString(taskFindFlags.org.id)
+	if b.org.id != "" {
+		id, err := influxdb.IDFromString(b.org.id)
 		if err != nil {
 			return err
 		}
 		filter.OrganizationID = id
 	}
 
-	if taskFindFlags.limit < 1 || taskFindFlags.limit > influxdb.TaskMaxPageSize {
+	if b.taskFindFlags.limit < 1 || b.taskFindFlags.limit > influxdb.TaskMaxPageSize {
 		return fmt.Errorf("limit must be between 1 and %d", influxdb.TaskMaxPageSize)
 	}
-	filter.Limit = taskFindFlags.limit
+	filter.Limit = b.taskFindFlags.limit
 
 	var tasks []*influxdb.Task
 
-	if taskFindFlags.id != "" {
-		id, err := influxdb.IDFromString(taskFindFlags.id)
+	if b.taskID != "" {
+		id, err := influxdb.IDFromString(b.taskID)
 		if err != nil {
 			return err
 		}
 
-		task, err := s.FindTaskByID(context.Background(), *id)
+		task, err := tskSvc.FindTaskByID(context.Background(), *id)
 		if err != nil {
 			return err
 		}
 
 		tasks = append(tasks, task)
 	} else {
-		tasks, _, err = s.FindTasks(context.Background(), filter)
+		tasks, _, err = tskSvc.FindTasks(context.Background(), filter)
 		if err != nil {
 			return err
 		}
 	}
 
-	return printTasks(
-		cmd.OutOrStdout(),
-		taskPrintOpts{
-			hideHeaders: taskPrintFlags.hideHeaders,
-			json:        taskPrintFlags.json,
-			tasks:       tasks,
-		},
-	)
+	return b.printTasks(taskPrintOpts{tasks: tasks})
+}
+
+type taskRerunFailedFlags struct {
+	before string
+	after  string
+}
+
+func (b *cmdTaskBuilder) taskRetryFailedCmd() *cobra.Command {
+	cmd := b.opts.newCmd("retry-failed", b.taskRetryFailedF, true)
+	cmd.Short = "Retry failed runs"
+	cmd.Aliases = []string{"rtf"}
+
+	b.org.register(b.opts.viper, cmd, false)
+	b.globalFlags.registerFlags(b.opts.viper, cmd)
+	registerPrintOptions(b.opts.viper, cmd, &b.taskPrintFlags.hideHeaders, &b.taskPrintFlags.json)
+	cmd.Flags().StringVarP(&b.taskID, "id", "i", "", "task ID")
+	cmd.Flags().StringVar(&b.taskRerunFailedFlags.before, "before", "", "runs before this time")
+	cmd.Flags().StringVar(&b.taskRerunFailedFlags.after, "after", "", "runs after this time")
+
+	return cmd
+}
+
+func (b *cmdTaskBuilder) taskRetryFailedF(*cobra.Command, []string) error {
+	if err := b.org.validOrgFlags(&flags); err != nil {
+		return err
+	}
+	tskSvc, _, err := b.svcFn()
+	if err != nil {
+		return err
+	}
+
+	taskIDPresent := b.taskID == ""
+
+	var failedRuns []*influxdb.Run
+	if !taskIDPresent {
+		failedRuns, err = b.getRunsOrg()
+	} else {
+		failedRuns, err = b.getFailedRunsTaskID()
+	}
+	if err != nil {
+		return err
+	}
+
+	for _, run := range failedRuns {
+		newRun, err := tskSvc.RetryRun(context.Background(), run.TaskID, run.ID)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Retry for task %s's run %s queued as run %s.\n", run.TaskID, run.ID, newRun.ID)
+	}
+	return nil
+
+}
+
+// helper function to find all failed runs based on taskID
```

Comment:
In general, top-level comments on types or functions in Go should begin with the name of the thing being documented, i.e.

```getFailedRunsTaskID is a helper function to find all failed runs based on taskID```

[Relevant bit from the Godoc introductory article on the Go blog:](https://blog.golang.org/godoc)

> Notice this comment is a complete sentence that begins with the name of the element it describes. This important convention allows us to generate documentation in a variety of formats, from plain text to HTML to UNIX man pages, and makes it read better when tools truncate it for brevity, such as when they extract the first line or sentence.

All of that said, you'll notice that when documented this way, the comment is redundant; `getFailedRunsTaskID` is pretty self-documenting, which is great! The only adjustment I might make would be `getFailedRunsByTaskID` for `getFailedRunsForTaskID`. This would help distinguish your use case of getting all runs for a task by ID from the case of getting the task ID for some given failed runs, with which `getFailedRunsTaskID` could be confused.